### DISCUSSION TITLE
Novice Questions

### DISCUSSION THREAD

**Discussion by AKHughes1994** [ratt-ru/solarkat#34] [Q&A] [answered]
Hi,

I'm going to take a crack at SolarKAT for some short-track MeerKAT data, and I have a few questions, 

1. Is the program PyPi compatible? The docs recommend cloning the git repository, but it might be pip installable.

2. I'm a stimela super-novice! But if I want to leave everything as default, so I'm running:

`stimela run solarkat.yaml solarkat`

Where do I feed in the ms file name, is there anything else critical that I change in the default files -- and which files should I modify? 

Thanks for the help,
Andrew

**Comment by o-smirnov** [comment]
@Victoria-Samboco can answer in more detail, but I think it is indeed on PyPI.

Looking at the recipe, the intended usage pattern is ``stimela run solarkat.yaml solarkat obs=label``, where label refers to a subsection in here such as "L1": https://github.com/ratt-ru/solarkat/blob/24759e4c7b075387019433191a8ec5540faa17c6/solarkat-pipeline/solarkat.yaml#L103

So you could make your own copy of ``solarkat.yaml`` and add your own section in there with your own label. It's also possible to pick a default but then override the individual parameters like so:

``stimela run solarkat.yaml solarkat obs=L1 ms=... dir_out=...``

I don't like the idea of baking obs-specific settings into the recipe file itself though, so I would suggest @Victoria-Samboco revise it to use this pattern:

```
assign_based_on:
  _include: defaults.yml
```

and then the setting can live in a separate user-supplied ``defaults.yml`` file.

**Comment by Victoria-Samboco** [comment, marked as answer]
> Hi,
> 
> I'm going to take a crack at SolarKAT for some short-track MeerKAT data, and I have a few questions,
> 
> 1. Is the program PyPi compatible? The docs recommend cloning the git repository, but it might be pip installable.

Hi, indeed, the version on PyPi is not updated, so for now, the best way to get solarkat is by cloning it from the git repository (I will work forward actualizing it on PyPI soon)

> 2. I'm a Stimela super-novice! But if I want to leave everything as default, so I'm running:
> 
> `stimela run solarkat.yaml solarkat`
> 
As @o-smirnov said you need to run it as: `stimela run solarkat.yaml solarkat obs=L1` if you want to specify which steps to run you can also do `stimela run solarkat.yaml solarkat obs=L1 -s image-1:image-4` or just `-s image-1` to run only that step and `-s image-1:` to run all steps from image-1.

> Where do I feed in the ms file name, is there anything else critical that I change in the default files -- and which files should I modify?
> 
you might probably have to modify the directories (for example where your MS is located). I will include a `solarkat-observations.yml` file for the observations' specifics, but for the actual version the obs specifics are defined in the solarkat.yml recipe itself. 

> Thanks for the help, Andrew
